crates/relayer-framework-test/src/relayer_mock/base/impls/chain.rs

```diff
@@ -0,0 +1,190 @@
+use async_trait::async_trait;
+use eyre::eyre;
+use std::vec;
+
+use crate::relayer_mock::base::error::Error;
+use crate::relayer_mock::base::types::chain::MockChainStatus;
+use crate::relayer_mock::base::types::events::{Event, WriteAcknowledgementEvent};
+use crate::relayer_mock::base::types::height::Height as MockHeight;
+use crate::relayer_mock::base::types::message::Message as MockMessage;
+use crate::relayer_mock::base::types::runtime::MockRuntimeContext;
+use crate::relayer_mock::base::types::state::State;
+use crate::relayer_mock::contexts::chain::MockChainContext;
+use ibc_relayer_framework::base::one_for_all::traits::chain::{
+    OfaBaseChain, OfaChainTypes, OfaIbcChain,
+};
+use ibc_relayer_framework::base::one_for_all::traits::runtime::OfaRuntimeContext;
+use ibc_relayer_framework::common::one_for_all::presets::MinimalPreset;
+
+impl OfaChainTypes for MockChainContext {
+    type Preset = MinimalPreset;
+
+    type Error = Error;
+
+    type Runtime = MockRuntimeContext;
+
+    type Height = MockHeight;
+
+    type Timestamp = MockHeight;
+
+    type Message = MockMessage;
+
+    type Signer = u128;
+
+    type RawMessage = String;
+
+    type Event = Event;
+
+    type ClientId = String;
+
+    type ConnectionId = String;
+
+    type ChannelId = String;
+
+    type PortId = String;
+
+    type Sequence = u128;
+
+    type WriteAcknowledgementEvent = WriteAcknowledgementEvent;
+
+    type ConsensusState = State;
+
+    type ChainStatus = MockChainStatus;
+}
+
+#[async_trait]
+impl OfaBaseChain for MockChainContext {
+    fn runtime(&self) -> &OfaRuntimeContext<MockRuntimeContext> {
+        self.runtime()
+    }
+
+    fn encode_raw_message(
+        message: &Self::Message,
+        _signer: &Self::Signer,
+    ) -> Result<Self::RawMessage, Self::Error> {
+        Ok(message.to_string())
+    }
+
+    // Only single messages are sent by the Mock Chain
+    fn estimate_message_len(_message: &Self::Message) -> Result<usize, Self::Error> {
+        Ok(1)
+    }
+
+    fn chain_status_height(status: &Self::ChainStatus) -> &Self::Height {
+        &status.height
+    }
+
+    fn chain_status_timestamp(status: &Self::ChainStatus) -> &Self::Timestamp {
+        &status.timestamp
+    }
+
+    fn try_extract_write_acknowledgement_event(
+        event: Self::Event,
+    ) -> Option<Self::WriteAcknowledgementEvent> {
+        match event {
+            Event::WriteAcknowledgment(h) => Some(WriteAcknowledgementEvent::new(h)),
+            _ => None,
+        }
+    }
+
+    /// If the message is a `SendPacket`, update the received packets,
+    /// and add a `RecvPacket` event to the returned array of events.
+    /// If the message is an `AckPacket`, update the received acknowledgment
+    /// packets.
+    /// If the message is an `UpdateClient` update the consensus state.
+    async fn send_messages(
+        &self,
+        messages: Vec<Self::Message>,
+    ) -> Result<Vec<Vec<Self::Event>>, Error> {
+        let mut res = vec![];
```

Comment:
Let's move the implementation body to a `process_messages` method in `MockChainContext`, so that we keep the trait implementation body simple.